File: src/Services/MovementService/MovementService.Application/DTOs/Response/MovementRequest/MovementRequestResponseDto.cs
using MovementService.Application.DTOs.Base;
using MovementService.Application.DTOs.Response.MovementRequestStatus;

namespace MovementService.Application.DTOs.Response.MovementRequest;

public class MovementRequestResponseDto : BaseDto
{
    public Guid ItemId { get; set; }  
    public Guid SourceWarehouseId { get; set; }
    public Guid DestinationWarehouseId { get; set; }
    public long Quantity { get; set; }
    public MovementRequestStatusResponseDto Status { get; set; }
    public DateTime RequestDate { get; set; }
    public Guid? ApprovedByUserId { get; set; }
}

File: src/Client/inventory-track/src/app/models/dto/movement-request/movement-request-response-dto.ts
import { MovementRequestStatusResponseDto } from './movement-request-status-response-dto';

export class MovementRequestResponseDto {
  id!: string; // Assuming BaseDto has an id property
  itemId!: string;
  sourceWarehouseId!: string;
  destinationWarehouseId!: string;
  quantity!: number;
  status!: MovementRequestStatusResponseDto;
  requestDate!: Date;
  approvedByUserId?: string;
}
